src/libraries/System.Private.CoreLib/src/System/IO/WindowsFileStreamStrategy.cs

```diff
@@ -32,6 +34,7 @@ internal abstract class WindowsFileStreamStrategy : FileStreamStrategy
         private readonly bool _isPipe;      // Whether to disable async buffering code.
 
         private long _appendStart; // When appending, prevent overwriting file.
+        private long? _length;
```

Comment:
what would be the difference in the allocated memory if we would use non-nullable `long` field and use negative values instead null to determine whether the field was initialized or not?

Reply:
This field doesn't actually allocate for being nullable; `Nullable<T>` is a struct, but it makes sense to use negatives instead.

Reply:
> This field doesn't actually allocate for being nullable

Is `sizeof(long) == sizeof(long?)`?

Reply:
> Is sizeof(long) == sizeof(long?)?

Nope, it is 8 and 16 respectively on my machine. I thought you were referring to heap allocations, I will prefer `long` for the upcoming pull request, thanks.